kit/src/main/java/com/oracle/javafx/scenebuilder/kit/fxom/FXMLPropertiesDisabler.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, Gluon and/or its affiliates.
+ * All rights reserved. Use is subject to license terms.
+ *
+ * This file is available and licensed under the following license:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the distribution.
+ *  - Neither the name of Oracle Corporation nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.oracle.javafx.scenebuilder.kit.fxom;
+
+class FXMLPropertiesDisabler {
+
+    /**
+     * 
```

Comment:
Very good idea, I have prepared something very similar in PR #396.
But now, where we have the nice OS enum, I'd go s slightly different way. Instead of providing the condition as a constructor argument, I'd pass in the OS instead. The FXMLPropertiesDisabler can then act according to what is required by OS or platform. Such as:

```java
private final OS os;

    /**
     * @param os Operating system where Scene Builder is executed
     */
    public FXMLPropertiesDisabler(OS os) {
        this.os = Objects.requireNonNull(os);
    }
```

When the class is used, it should be constructed like that:

```java
  new FXMLPropertiesDisabler(OS.get());
  new FXMLPropertiesDisabler();  // this could be the default constrcutor where OS.get() is used in
```

And for the test:

```java
new FXMLPropertiesDisabler(OS.MAC);
new FXMLPropertiesDisabler(OS.WINDOWS);
new FXMLPropertiesDisabler(OS.LINUX);
...
```

What do you think?

Reply:
I limited the scope of the first occurrence of the OS enum in this PR as I thought its too much of a change.

Reply:
A Predicate is more flexible, but so far she only use the OS itself, good enough for now.